src/v/config/configuration.cc

```diff
@@ -1110,7 +1110,8 @@ configuration::configuration()
   , sasl_mechanisms(
       *this,
       "sasl_mechanisms",
-      "A list of supported SASL mechanisms. `SCRAM` and `GSSAPI` are allowed.",
+      "A list of supported SASL mechanisms. `SCRAM`, `GSSAPI`, and "
+      "`OAUTHBEARER` are allowed.",
```

Comment:
chore: Update cover letter of PR in the docs section to indicate the update to this config property

Reply:
Done